src/Compilers/CSharp/Test/Semantic/Semantics/ReadOnlyStructsTests.cs

```diff
@@ -1087,9 +1086,13 @@ public struct S
 ";
             var comp = CreateCompilation(csharp);
             comp.VerifyDiagnostics(
-                // (5,32): error CS0106: The modifier 'readonly' is not valid for this item
+                // (5,32): error CS8657: Static member 'S.M()' cannot be 'readonly'.
                 //     public static readonly int M()
-                Diagnostic(ErrorCode.ERR_BadMemberFlag, "M").WithArguments("readonly").WithLocation(5, 32));
+                Diagnostic(ErrorCode.ERR_StaticMemberCantBeReadOnly, "M").WithArguments("S.M()").WithLocation(5, 32));
+
+            var method = comp.GetMember<NamedTypeSymbol>("S").GetMethod("M");
```

Comment:
Please also test the public API (unless you haven't hooked up the public version yet)

Reply:
Public API isn't exposed yet--we're also not positive about whether to expose `IsDeclaredReadOnly`, `IsEffectivelyReadOnly`, or both.